src-java/server42/server42-control-storm-topology/src/main/java/org/openkilda/server42/control/topology/storm/bolt/flow/FlowHandler.java

```diff
@@ -0,0 +1,142 @@
+    }
+
+    @Override
+    protected void handleInput(Tuple input) throws Exception {
+        String source = input.getSourceComponent();
+        if (Router.BOLT_ID.equals(source)) {
+            handleRouterCommand(input);
+        } else {
+            unhandledInput(input);
+        }
+    }
+
+    @Override
+    protected void handleException(Exception error) throws Exception {
+        try {
+            super.handleException(error);
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void declareOutputFields(OutputFieldsDeclarer outputFieldsDeclarer) {
+        outputFieldsDeclarer.declareStream(STREAM_CONTROL_COMMANDS_ID, STREAM_CONTROL_COMMANDS_FIELDS);
+    }
+
+    private void handleRouterCommand(Tuple input) throws PipelineException {
+        handleCommand(input, Router.FIELD_ID_COMMAND);
+    }
+
+    private void handleCommand(Tuple input, String fieldName) throws PipelineException {
+        FlowCommand command = pullValue(input, fieldName, FlowCommand.class);
+        command.apply(this);
+    }
+
+    public void processActivateFlowMonitoring(String flowId, FlowEndpointPayload flow, boolean isForward) {
+        flowRttService.activateFlowMonitoring(flowId, flow.getDatapath(), flow.getPortNumber(),
+                flow.getVlanId(), isForward);
+    }
+
+    public void processDeactivateFlowMonitoring(SwitchId switchId, String flowId) {
+        // no logic just repack
+        notifyDeactivateFlowMonitoring(switchId, flowId);
+    }
+
+    public void processActivateFlowMonitoringOnSwitch(SwitchId switchId) {
+        flowRttService.activateFlowMonitoringForSwitch(switchId);
+    }
+
+    public void processDeactivateFlowMonitoringOnSwitch(SwitchId switchId) {
+        // no logic just repack
+        notifyDeactivateFlowMonitoring(switchId);
+    }
+
+    @Override
+    public void notifyActivateFlowMonitoring(String flowId, SwitchId switchId, Integer port, Integer vlan,
+                                             boolean isForward) {
+        AddFlow addFlow = AddFlow.builder()
+                .flowId(flowId)
+                .port(port)
+                .tunnelId(vlan.longValue())
+                .direction(isForward ? FlowDirection.FORWARD : FlowDirection.REVERSE)
+                .headers(buildHeader())
+                .build();
+
+        emit(STREAM_CONTROL_COMMANDS_ID, getCurrentTuple(), new Values(switchId.toString(), addFlow));
+    }
+
+    @Override
+    public void notifyDeactivateFlowMonitoring(SwitchId switchId, String flowId) {
+        RemoveFlow removeFlow = RemoveFlow.builder()
+                .flowId(flowId)
+                .headers(buildHeader())
+                .build();
+        emit(STREAM_CONTROL_COMMANDS_ID, getCurrentTuple(), new Values(switchId.toString(), removeFlow));
+    }
+
+    private void notifyDeactivateFlowMonitoring(SwitchId switchId) {
+        ClearFlows clearFlows = new ClearFlows();
```

Comment:
Why there is no headers(correlation-id) here?

Reply:
done